packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -663,22 +671,113 @@ class RenderWrap extends RenderBox
 
   (_AxisSize childrenSize, List<_RunMetrics> runMetrics) _computeRuns(BoxConstraints constraints, ChildLayouter layoutChild) {
     assert(firstChild != null);
-    final (BoxConstraints childConstraints, double mainAxisLimit) = switch (direction) {
-      Axis.horizontal => (BoxConstraints(maxWidth: constraints.maxWidth), constraints.maxWidth),
-      Axis.vertical => (BoxConstraints(maxHeight: constraints.maxHeight), constraints.maxHeight),
+    final (BoxConstraints childConstraintsLoose,
+           BoxConstraints childConstraintsTight,
+           double mainAxisLimit,
+           double Function(RenderBox child, double crossAxisSize) getChildMinIntrinsicMainAxisExtent,
+           BoxConstraints Function(double maxSize) childConstraintsFittingLooseInRun,
+           BoxConstraints Function(double maxSize) childConstraintsFittingTightInRun) = switch (direction) {
+      Axis.horizontal => (
+           BoxConstraints(maxWidth: constraints.maxWidth),
+           BoxConstraints(minWidth: constraints.maxWidth, maxWidth: constraints.maxWidth),
+           constraints.maxWidth,
+           (RenderBox child, double crossAxisSize) => child.getMinIntrinsicWidth(crossAxisSize),
+           (double maxSize) => BoxConstraints(maxWidth: maxSize),
+           (double maxSize) => BoxConstraints(minWidth: maxSize, maxWidth: maxSize)),
+      Axis.vertical => (
+           BoxConstraints(maxHeight: constraints.maxHeight),
+           BoxConstraints(minHeight: constraints.maxHeight, maxHeight: constraints.maxHeight),
+           constraints.maxHeight,
+           (RenderBox child, double crossAxisSize) => child.getMinIntrinsicHeight(crossAxisSize),
+           (double maxSize) => BoxConstraints(maxHeight: maxSize),
+           (double maxSize) => BoxConstraints(minHeight: maxSize, maxHeight: maxSize)),
```

Comment:
I'm not sure if this could be improved. For me it's fine, but I can see someone having a bad time reading this block. Esp. the switch is quite hard to spot.